### PR TITLE
INTYGFV-14572: added functionality to resend the cryptokey.

### PR DESCRIPTION
**Pull request by IamFriberg**
Added functionality to resend the cryptokey.

### CODE REVIEW

File: web/build.gradle.kts [hunk 1]
```diff
@@ -150,7 +150,7 @@ tasks {
   bootRun {
     systemProperty("dev.http.port", devPort)
     systemProperty("dev.http.port.internal", devPortInternal)
-    systemProperty("spring.profiles.active", "dev, fake, caching-enabled, it-stub, wc-stub, pp-stub, ts-stub")
+    systemProperty("spring.profiles.active", "dev, fake, caching-enabled, it-stub, wc-stub, pp-stub")
```
Comment (mikeyOne) [inline web/build.gradle.kts:153]: Perhaps we should keep the ts-stub as default?

File: web/src/main/java/se/inera/intyg/intygsadmin/web/service/TerminationServiceImpl.java [hunk 2]
```diff
@@ -106,10 +139,20 @@ private Comparator<DataExportResponse> getComparator(String sortColumn, Directio
             .thenComparing(DataExportResponse::getCreated, Comparator.reverseOrder());
     }
 
+    /**
+     * Check direktion to sert
```
Comment (mikeyOne) [inline web/src/main/java/se/inera/intyg/intygsadmin/web/service/TerminationServiceImpl.java:143]: Spelling in description :)

File: web/src/main/java/se/inera/intyg/intygsadmin/web/stubs/TerminationRestServiceStub.java [hunk 3]
```diff
@@ -48,17 +48,161 @@ public TerminationRestServiceStub() {
...
+        final var dataExportEntity3 = new DataExportResponse();
+        dataExportEntity3.setTerminationId(UUID.randomUUID());
+        dataExportEntity3.setCreated(LocalDateTime.now().minusMinutes(2L));
+        dataExportEntity3.setStatus("Intyg hämtade");
+        dataExportEntity3.setCreatorName("Sten Stensson");
+        dataExportEntity3.setPhoneNumber("08-33333333");
+        dataExportEntity3.setEmailAddress("sten@stensson.se");
+        dataExportEntity3.setOrganizationNumber("654321-1111");
+        dataExportEntity3.setPersonId("19770523-0001");
+        dataExportEntity3.setHsaId("SE2321000999-esc");
```
Comment (mikeyOne) [inline web/src/main/java/se/inera/intyg/intygsadmin/web/stubs/TerminationRestServiceStub.java:69]: Nice to have all statuses represented in stub but perhaps introduce variation between the posts (also in hsaIds, email, personId etc.) to make it more real, for example when sorting on different columns...

File: web/src/main/java/se/inera/intyg/intygsadmin/web/integration/TerminationRestServiceImpl.java [hunk 4]
```diff
@@ -80,4 +94,15 @@ public String eraseDataExport(String terminationId) {
             throw e;
         }
     }
+
+    @Override
+    public String resendDataExportKey(String terminationId) {
+        String url = terminationServiceUrl + "api/v1/terminations/" + terminationId + "/resendpassword";
```
Comment (mikeyOne) [inline web/src/main/java/se/inera/intyg/intygsadmin/web/integration/TerminationRestServiceImpl.java:100]: Description for this method?

File: web/src/main/java/se/inera/intyg/intygsadmin/web/service/TerminationServiceImpl.java [hunk 5]
```diff
@@ -86,17 +96,40 @@ public DataExportResponse createDataExport(CreateDataExportDTO createDataExportD
...
+     * @return
+     */
     @Override
     public String eraseDataExport(String terminationId) {
         return terminationRestService.eraseDataExport(terminationId);
     }
 
+    @Override
+    public String resendDataExportKey(String terminationId) {
+        return terminationRestService.resendDataExportKey(terminationId);
```
Comment (mikeyOne) [inline web/src/main/java/se/inera/intyg/intygsadmin/web/service/TerminationServiceImpl.java:111]: Description for this method?

**Review by mikeyOne — CHANGES_REQUESTED** [review]

File: web/build.gradle.kts (hunk 1 above)
Comment (IamFriberg) [reply web/build.gradle.kts:153]: Commited it by mistake :)
Fixed!

File: web/src/main/java/se/inera/intyg/intygsadmin/web/service/TerminationServiceImpl.java (hunk 2 above)
Comment (IamFriberg) [reply web/src/main/java/se/inera/intyg/intygsadmin/web/service/TerminationServiceImpl.java:143]: Fixed!

File: web/src/main/java/se/inera/intyg/intygsadmin/web/stubs/TerminationRestServiceStub.java (hunk 3 above)
Comment (IamFriberg) [reply web/src/main/java/se/inera/intyg/intygsadmin/web/stubs/TerminationRestServiceStub.java:69]: Fixed!

File: web/src/main/java/se/inera/intyg/intygsadmin/web/integration/TerminationRestServiceImpl.java (hunk 4 above)
Comment (IamFriberg) [reply web/src/main/java/se/inera/intyg/intygsadmin/web/integration/TerminationRestServiceImpl.java:100]: Fixed

File: web/src/main/java/se/inera/intyg/intygsadmin/web/service/TerminationServiceImpl.java (hunk 5 above)
Comment (IamFriberg) [reply web/src/main/java/se/inera/intyg/intygsadmin/web/service/TerminationServiceImpl.java:111]: Fixed

**Review by mikeyOne — APPROVED** [review]

**Pull request merged**